### PR TITLE
Documentation Crawler

### PR DESCRIPTION
**Pull request by Nolski**
The following is a python scrapy web crawler which should generate our end result API command json files which we can use to do _some_ automatic conversion between library commands.

The json files are structured as follows:

```json
[
  {
    "code": "torch.command(a, b=None, c=2)",
    "function_name": "command",
    "args": ["a"],
    "kwargs": [["b", "None"], ["c", "2"]]
  },
  ...
]
```

### CODE REVIEW

**Comment by Nolski** [comment]
@cereallarceny feel free to give this a look and see if the json schema looks good to you.

**Comment by cereallarceny** [comment]
This tentatively looks good to me @nolski.  Quick question: would it be better to have the command name be the key of the object?  I see you're generating this as an array.  Would it be better to be a big object with the command name as the key?  Just a thought - although I don't have a strong opinion and will trust your judgement here.

Likewise, what will be our strategy for versioning?

**Comment by Nolski** [comment]
@cereallarceny Just finished up this PR and I think it's ready to review for now.

With regards to having the whole list be turned into an object, I considered it, but decided against is mostly in the name of simplicity. It's mostly formatted this way as Scrapy prefers to store items as a list. I figure mapping over the list shouldn't be too slow given the size each file but if that ends up being a problem, I can follow up and fix in a separate PR.

I currently don't have versioning of the API docs in this PR. I opted not to include it as it could make a good first issue and didn't want to expand the scope of this PR too much.

File: docs-crawler/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Threepio Documentation Crawler
+
+This documentation web crawler is used to generate a set of static JSON files which we can use to help perform certain automatic translations between PyTorch, Tensorflow, and Tensorflow.js.
+
+## Installation
```
Comment (Nolski) [inline docs-crawler/README.md:5]: Feel free to pull down the branch and install yourself. I know poetry is a bit less common than pip/venv but it seems fairly easy to install & use.

File: docs-crawler/docs/docs/spiders/tf_spider.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import re
```
Comment (Nolski) [inline docs-crawler/docs/docs/spiders/tf_spider.py:1]: Most of the files here are generated by Scrapy, however these "spider" files are the main components doing web scraping of the docs and formatting them into json blobs or "Items"

**Comment by Nolski** [comment]
I'm gonna go ahead and merge this for now since it's all so early stage. 

@cereallarceny if you have any feedback, leave it here and I'll re-open and merge the suggested changes or open new tickets accordingly.

**Pull request merged**